src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -407,7 +403,7 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
                 }
                 else
                 {
-                    if (ClientCertificates.Count == 0)
+                    if (_clientCertificates == null || _clientCertificates.Count == 0)
```

Comment:
Since the if and else have code in common, the common could be moved out of the branch to

```C#
if (NetEventSource.IsEnabled) NetEventSource.Log.NoDelegateNoClientCert(this);
if (ClientCertificates.Count == 0)
{
    sessionRestartAttempt = true;
}
```
(Assuming `NoDelegateNoClientCert` doesn't side-effect on `ClientCertificates` which I haven't checked. Not a big impact, but if one was going to refactor the if, I'd say do that rather than flipping it.